main.py

```diff
@@ -38,15 +38,16 @@ def knapsack_bqm(costs, weights, weight_capacity):
     # Number of objects
     x_size = len(costs)
 
-    # Lucas's algorithm introduces additional slack variables to handle
-    # the inequality. max_y_index indicates the maximum index in the y
-    # sum; hence the number of slack variables.
-    max_y_index = ceil(log(weight_capacity))
```

Comment:
@mcfarljm, is the change necessary, from `ceil` to `floor`? I would like to understand it.

Reply:
Yes, it is necessary with the log2 formulation.  Please see the message for 0fbfcf7, where I described an example.